spyql/parser.py

```diff
@@ -127,6 +127,12 @@ def pythonize(s):
         "count_distinct_agg(tuple(_values))", s
     )
 
+    # universal access syntax
+    # `.column` is converted to `row.column`
+    s = re.compile(r"(^|[^\)\]\}\.\w\"\'\`])\.(?=[A-Za-z_])").sub(r"\1row.", s)
```

Comment:
❓ I can't make sense of this first group in the regex... You tested it and it's working, so I'm going to approve it 😅

Reply:
Basically, it is a list of all characters that are not allowed before a dot (in order for the expression to be considered for this use case). e.g. in `(1,2).count(2)` the dot follows a `)` and so it would refer to an attribute/method of a tuple, while in `(.a_col)` there is no relation between the dot and `(`. So `)` is in the regex has a forbidden prefix of `.` while  `(` is not.